mmv1/third_party/terraform/services/logging/resource_logging_project_sink.go

```diff
@@ -39,6 +39,11 @@ func ResourceLoggingProjectSink() *schema.Resource {
 		Default:     false,
 		Description: `Whether or not to create a unique identity associated with this sink. If false (the default), then the writer_identity used is serviceAccount:cloud-logs@system.gserviceaccount.com. If true, then a unique service account is created and used for this sink. If you wish to publish logs across projects, you must set unique_writer_identity to true.`,
 	}
+	schm.Schema["custom_writer_identity"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
```

Comment:
`unique_writer_identity` has a default value true, and if `custom_writer_identity` is provided, the `unique_writer_identity` will also be set to true. So I think `RequiredWith` is not needed here?

Reply:
It looks like `unique_writer_identity` has a default value of `false` in this file.

Reply:
@pengq-google when testing with REST API, i.e
```
https://logging.googleapis.com/v2/projects/eng-xxx/sinks/xxx-test-sink?uniqueWriterIdentity=true&customWriterIdentity=serviceAccount:testconfigconnector@eng-xxx.iam.gserviceaccount.com'](https://logging.googleapis.com/v2/projects/xxx/sinks/xxx-test-sink?uniqueWriterIdentity=true&customWriterIdentity=serviceAccount:testconfigconnector@xxx.iam.gserviceaccount.com%27
```
If I don't specify the `uniqueWriterIdentity` in query param, get error:
```
{
  "error": {
    "code": 400,
    "message": "The destination project must be the same as the sink's project. Use uniqueWriterIdentity=true to allow cross project exports.",
    "status": "INVALID_ARGUMENT"
  }
}
``` 
Thus, I think that the `RequiredWith` is needed here for Terraform. And it should always be true when used along with customWriterIdentity. 
If that's the case, maybe we can safely set the uniqueWriterIdentity=true regardless user's input? or we let the backend bubbles up the error as above? @melinath thoughts?

Reply:
I realized that the: only if your log sinks are in different projects, then you can use custom-managed service account. In order to set log sink from a different projects, users have to set the `uniqueWriterIdentity=true`. So it is a indirect dependency for this case. But seems `RequiredWith` is needed.

Reply:
1) I thought with terraform setting the unique_writer_identity default to true it should already set it. seems I am wrong.
2) we have a cl submitted so that with provided custom_writer_identity, value of unique_writer_identity will be ignored, this cl will be deployed to prod quite soon.

Reply:
1. The default was `false` in 4.X. It looks like the default was updated to `true` in 5.X, and this PR has now been updated to include that change. I'm not sure how that would play with `RequiredWith`.
2. If the API behavior is changing to not require them together then we definitely don't need `RequiredWith` here.